subscription_oca/models/sale_order.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 Domatix - Carlos Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from collections import defaultdict
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import Command, api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    subscription_ids = fields.One2many(
+        comodel_name="sale.subscription",
+        inverse_name="sale_order_id",
+        string="Subscriptions",
+    )
+    subscriptions_count = fields.Integer(compute="_compute_subscriptions_count")
+    order_subscription_id = fields.Many2one(
+        comodel_name="sale.subscription", string="Subscription"
+    )
+
+    @api.depends("subscription_ids")
+    def _compute_subscriptions_count(self):
+        for record in self:
+            record.subscriptions_count = len(record.subscription_ids)
+
+    def action_view_subscriptions(self):
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "sale.subscription",
+            "domain": [("id", "in", self.subscription_ids.ids)],
+            "name": self.name,
+            "view_mode": "list,form",
+        }
+
+    def get_next_interval(self, type_interval, interval):
+        date_start = date.today()
+        date_start += relativedelta(**{type_interval: interval})
+        return date_start
+
+    def create_subscription(self, lines, subscription_tmpl):
+        self.ensure_one()
+        subscription_lines = []
```

Comment:
You can refactor like this:
`subscription_lines = [Command.create(line.get_subscription_line_values()) for line in lines]`.
And if I correctly understand the code, this should be moved into the `if` part.